scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/context/TaskContext.java

```diff
@@ -62,21 +62,28 @@
 
     private final String progressFilePath;
 
+    private final String nodeUrl;
+
+    private final String nodeName;
+
     public TaskContext(ExecutableContainer executableContainer, TaskLauncherInitializer initializer,
             TaskResult[] previousTasksResults, NodeDataSpacesURIs nodeDataSpacesURIs, String progressFilePath,
-            String currentNodeHostname) throws NodeException {
+            NodeInfo nodeInfo) throws NodeException {
         this(executableContainer,
              initializer,
              previousTasksResults,
              nodeDataSpacesURIs,
              progressFilePath,
-             currentNodeHostname,
+             nodeInfo.getCurrentNodeHostname(),
+             nodeInfo.getCurrentNodeUrl(),
+             nodeInfo.getCurrentNodeName(),
              null);
     }
 
     public TaskContext(ExecutableContainer executableContainer, TaskLauncherInitializer initializer,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 9 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)